source/grid/tria.cc

```diff
@@ -5716,7 +5716,10 @@ namespace internal
                     const std::array<unsigned int, 12> line_indices =
                       TriaAccessorImplementation::Implementation::
                         get_line_indices_of_cell(*hex);
-                    for (unsigned int l = 0; l < hex->n_lines(); ++l)
+                    // avoid a compiler warning by fixing the max number of
+                    // loop iterations to 12
+                    const unsigned int n_lines = std::min(hex->n_lines(), 12u);
+                    for (unsigned int l = 0; l < n_lines; ++l)
```

Comment:
@kronbichler What do you think? This was what you said last year: https://github.com/dealii/dealii/issues/14263#issuecomment-1244986296